appcelerator/titanium_mobile
# feat: add Ti.UI.OptionBar

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28317

**Summary:**
- On iOS, this shows a segmented control.
- On Android, this shows a material toggle button group.
  * The `layout` property can be set to `"horiztonal"` (the default) or `"vertical"`.

**Examples:**
<img width="295" alt="OptionBar-iOS" src="https://user-images.githubusercontent.com/26234558/109606302-4e98df00-7adb-11eb-86ad-f533ec2f3916.png"> <img width="244" alt="OptionBar-Android-Horizontal" src="https://user-images.githubusercontent.com/26234558/109606176-201b0400-7adb-11eb-844f-a4f2568cb515.png"> <img width="244" alt="OptionBar-Android-Vertical" src="https://user-images.githubusercontent.com/26234558/109606187-23ae8b00-7adb-11eb-8bdb-6e9f9aa76183.png">

---
**Test:**
1. Create a Classic Titanium project. (Use the template that creates a TabGroup.)
2. Use the `OptionBarTest.js` as your `app.js` from [TIMOB-28317](https://jira.appcelerator.org/browse/TIMOB-28317).
3. Add the below <application/> element to the "tiapp.xml" file.
4. Build and run on Android.
5. Verify the app looks like the above screenshot.
6. Tap on the 2nd button of each bar and verify the following gets logged.
`@@@ button click index: 1`
7. Repeat steps 5-6 on iOS.
8. Uncomment the `// layout: "vertical",` lines in all 3 spots of the code.
9. Repeat steps 5-6 on Android and verify it looks like the above vertical buttons.

```xml
<ti:app>
	<android xmlns:android="http://schemas.android.com/apk/res/android">
		<manifest>
			<application android:theme="@style/Theme.MaterialComponents.Light.DarkActionBar"/>
		</manifest>
	</android>
</ti:app>
```
